File: C#/Pascal's Triangle/Pascal's Triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        IList<IList<int>> triangle=new List<IList<int>>();
        if(numRows<=0) return triangle;
        IList<int> top=new List<int>();
        top.Add(1);
        triangle.Add(top);
        for(int i=1;i<numRows;i++){
            triangle.Add(Level(triangle[triangle.Count-1]));
        }
        return triangle;
    }
    
    private IList<int> Level(IList<int> upLevel){
        IList<int> level=new List<int>();
        for(int i=0;i<upLevel.Count;i++){
            int val=(i==0||i==upLevel.Count)?0:upLevel[i-1];
            level.Add(val+upLevel[i]);
        }
        level.Add(1);
        return level;
    }
}

File: Python/Pascal's Triangle/Pascal's Triangle.py
class Solution:
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        if(numRows<=0):
            return []
        result=[[1]]
        for i in range(numRows-1):
            result.append(self.level(result[-1]))
        return result
    
    def level(self,upLevel):
        thisLevel=[]
        newLevel=upLevel[:];
        newLevel.append(0)
        newLevel=newLevel[::-1]
        newLevel.append(0)
        for i in range(len(newLevel)-1):
            thisLevel.append(newLevel[i]+newLevel[i+1])
        return thisLevel